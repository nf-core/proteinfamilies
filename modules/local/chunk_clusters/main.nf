process CHUNK_CLUSTERS {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/43/438649357e4dcaab676b1bff95e3aace1decb36a658d6257869a641155867e0c/data' :
        'community.wave.seqera.io/library/pip_pyfastx:c1d255a74c4291f8' }"

    input:
    tuple val(meta) , path(clustering)
    tuple val(meta2), path(sequences)
    val(size_threshold)

    output:
    tuple val(meta), path("chunked_fasta/*"), emit: fasta_chunks
    path "versions.yml"                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def is_compressed = sequences.getName().endsWith(".gz") ? true : false
    def fasta_name    = sequences.name.replace(".gz", "")
    """
    if [ "$is_compressed" == "true" ]; then
        gzip -c -d $sequences > $fasta_name
    fi

    chunk_clusters.py \\
        --clustering ${clustering} \\
        --sequences ${fasta_name} \\
        --threshold ${size_threshold} \\
        --out_folder chunked_fasta

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version 2>&1 | sed 's/Python //g')
        biopython: \$(python -c "import importlib.metadata; print(importlib.metadata.version('biopython'))")
    END_VERSIONS
    """

    stub:
    """
    mkdir -p chunked_fasta
    touch chunked_fasta/1.fasta

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version 2>&1 | sed 's/Python //g')
        biopython: \$(python -c "import importlib.metadata; print(importlib.metadata.version('biopython'))")
    END_VERSIONS
    """

}
