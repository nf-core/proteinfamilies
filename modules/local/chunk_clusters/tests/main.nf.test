nextflow_process {

    name "Test Process CHUNK_CLUSTERS"
    script "../main.nf"
    process "CHUNK_CLUSTERS"

    test("proteinfamilies - tsv") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'test_tsv' ], // meta map
                    file(params.pipelines_testdata_base_path + 'proteinfamilies/test_data/modules/clustering.tsv', checkIfExists: true)
                ])
                input[1] = Channel.of([
                    [ id: 'test_fasta' ], // meta map
                    file(params.pipelines_testdata_base_path + 'proteinfamilies/test_data/mgnifams_input_small.fa', checkIfExists: true)
                ])
                input[2] = 5
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("proteinfamilies - tsv - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'test_tsv' ], // meta map
                    file(params.pipelines_testdata_base_path + 'proteinfamilies/test_data/modules/clustering.tsv', checkIfExists: true)
                ])
                input[1] = Channel.of([
                    [ id: 'test_fasta' ], // meta map
                    file(params.pipelines_testdata_base_path + 'proteinfamilies/test_data/mgnifams_input_small.fa', checkIfExists: true)
                ])
                input[2] = 5
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
