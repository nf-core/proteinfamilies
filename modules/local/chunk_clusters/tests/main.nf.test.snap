{
    "proteinfamilies - tsv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_tsv"
                        },
                        [
                            "1.fasta:md5,f5e1235669a420aa4506197a041c2c21",
                            "2.fasta:md5,44337cbbd316c799ccbf73119adb99bb",
                            "3.fasta:md5,1144e5e8ae2e435581622ceca6d4f197"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,15841bc0b0db3156bdce3d636ff151f8"
                ],
                "fasta_chunks": [
                    [
                        {
                            "id": "test_tsv"
                        },
                        [
                            "1.fasta:md5,f5e1235669a420aa4506197a041c2c21",
                            "2.fasta:md5,44337cbbd316c799ccbf73119adb99bb",
                            "3.fasta:md5,1144e5e8ae2e435581622ceca6d4f197"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,15841bc0b0db3156bdce3d636ff151f8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-26T13:57:26.186888"
    },
    "proteinfamilies - tsv - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_tsv"
                        },
                        "1.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,15841bc0b0db3156bdce3d636ff151f8"
                ],
                "fasta_chunks": [
                    [
                        {
                            "id": "test_tsv"
                        },
                        "1.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,15841bc0b0db3156bdce3d636ff151f8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-26T13:58:21.283468"
    }
}